### PR TITLE
fixes (#7843) Hide the first run checkbox for desktop

### LINKED ISSUE #7843: Hide the first run checkbox for desktop
Becuase the first run checkbox only applies for mobile experiments we should hide it altogether for desktop experiments to prevent confusion.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-2847)

### PR DESCRIPTION
**Pull request by cherish2003**
### Because
Hide the first run checkbox for desktop

### This commit

- Added logic not to show first run checkbox for desktop
- Modified the tests and written new tests for `isfirstrun`
- fixes #7843

### CODE REVIEW

**Comment by cherish2003** [comment]
Hello @yashikakhurana ma'am, I have modified the changes as per your advice , Please let me know if any more changes are required , thanks

**Comment by cherish2003** [comment]
@yashikakhurana ma'am can you inform why this checks are failing suddenly after updating the branch ?

**Comment by yashikakhurana** [comment]
@cherish2003 it is good now after updating

**Comment by cherish2003** [comment]
@yashikakhurana ma'am is there any more changes needed

File: experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.test.tsx [hunk 1]
```diff
@@ -121,7 +121,6 @@ describe("FormAudience", () => {
     );
 
     expect(screen.getByTestId("isSticky")).toBeChecked();
-    expect(screen.getByTestId("isFirstRun")).toBeChecked();
```
Comment (yashikakhurana) [inline experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.test.tsx:124]: instead of removing this you can check this element is not present

File: experimenter/experimenter/nimbus-ui/src/hooks/useCommonForm/index.test.tsx [hunk 2]
```diff
@@ -149,7 +149,7 @@ describe("hooks/useCommonForm", () => {
     it("FormAudience", () => {
       render(<AudienceSubject />);
       const filteredAudienceFieldNames = audienceFieldNames.filter(
-        (e) => e !== "languages",
+        (e) => e !== "languages" && e !== "isFirstRun",
```
Comment (yashikakhurana) [inline experimenter/experimenter/nimbus-ui/src/hooks/useCommonForm/index.test.tsx:152]: same question- why this change?

**Review by yashikakhurana — CHANGES_REQUESTED** [review]

File: experimenter/experimenter/nimbus-ui/src/hooks/useCommonForm/index.test.tsx (hunk 2 above)
Comment (cherish2003) [reply experimenter/experimenter/nimbus-ui/src/hooks/useCommonForm/index.test.tsx:152]: This change is not to check `isFirstRun` for the desktop which is the same case with `languages` so i added it here

File: experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.test.tsx (hunk 1 above)
Comment (yashikakhurana) [reply experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.test.tsx:124]: yeah so if its a desktop it should not render checkbox,

Comment (yashikakhurana) [reply experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.test.tsx:124]: so instead of removing line, modify it as it is not present

Comment (cherish2003) [reply experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.test.tsx:124]: tried it but showing error element not found by getByTestid ( This is because in Desktop `isfirstrun` is not present to get it so which results in this error )

Comment (yashikakhurana) [reply experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.test.tsx:124]: you can do something like this
    expect(testelement).not.toBeInTheDocument();

Comment (cherish2003) [reply experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.test.tsx:124]: @yashikakhurana i have made changes as you said once check it and let me know if any further changes are required

File: experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.test.tsx [hunk 3]
```diff
@@ -121,7 +121,7 @@ describe("FormAudience", () => {
...
+    expect(screen.queryByText("isFirstRun")).not.toBeInTheDocument();
```
Comment (yashikakhurana) [inline experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.test.tsx:124]: text will not be `isFirstRun`- it should be `First Run Experiment` or query by id

**Review by yashikakhurana — CHANGES_REQUESTED** [review]

File: experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.test.tsx (hunk 3 above)
Comment (cherish2003) [reply experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.test.tsx:124]: @yashikakhurana as i told above querying by id results in error so i would rather change the text inside `queryByText` as per your mentions

Comment (cherish2003) [reply experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.test.tsx]: @yashikakhurana i have the made changes have a look at it

File: experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.test.tsx [hunk 4]
```diff
@@ -453,15 +453,15 @@ describe("FormAudience", () => {
...
         experiment={{
           ...MOCK_EXPERIMENT,
-          application: NimbusExperimentApplicationEnum.DESKTOP,
+          application: NimbusExperimentApplicationEnum.FENIX,
           channel: NimbusExperimentChannelEnum.NIGHTLY,
-          targetingConfigSlug: "WIN_ONLY",
+          targetingConfigSlug: "MOBILE_ONLY",
           targetingConfig: [
             {
-              label: "Win Only",
```
Comment (yashikakhurana) [inline experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.test.tsx:461]: since you have removed all the windows targeting to mobile, I would suggest adding a test with windows

**Review by yashikakhurana — CHANGES_REQUESTED** [review]

**Review by yashikakhurana — CHANGES_REQUESTED** [review]

File: experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.test.tsx [hunk 5]
```diff
@@ -726,6 +807,19 @@ describe("FormAudience", () => {
     expect(screen.queryByTestId("locales")).not.toBeInTheDocument();
   });
 
+  it("enables isFirstRun for mobile", async () => {
```
[the remark this answers is not in the document]
Comment (cherish2003) [reply experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.test.tsx]: @yashikakhurana changed the test name as per your required

File: experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.test.tsx [hunk 6]
```diff
@@ -739,6 +833,18 @@ describe("FormAudience", () => {
     expect(screen.queryByTestId("languages")).not.toBeInTheDocument();
     expect(screen.queryByTestId("locales")).toBeInTheDocument();
   });
+  it("disables isFirstRun for desktop", async () => {
```
[the remark this answers is not in the document]
Comment (cherish2003) [reply experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.test.tsx]: @yashikakhurana changed the test name as per your required

File: experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.test.tsx [hunk 7]
```diff
@@ -833,7 +833,7 @@ describe("FormAudience", () => {
     expect(screen.queryByTestId("languages")).not.toBeInTheDocument();
     expect(screen.queryByTestId("locales")).toBeInTheDocument();
   });
-  it("disables isFirstRun for desktop", async () => {
+  it("isFirstRun does not renders for mobile application", async () => {
```
Comment (yashikakhurana) [inline experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.test.tsx:836]: it should be desktop application

**Review by yashikakhurana — CHANGES_REQUESTED** [review]

File: experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.test.tsx (hunk 7 above)
Comment (cherish2003) [reply experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.test.tsx]: @yashikakhurana sorry for the typo fixed it 👍🏻

**Review by yashikakhurana — APPROVED** [review]
Thank you @cherish2003 great tests 👏
